### PR TITLE
Add Played Tiles Interaction

### PR DESCRIPTION
**Pull request by yummyblabla**
- [x] Render played tiles for each player

- [x] Add Interactions to interaction container (triplet, consecutive, win) // placeholders until meld validation is implemented

- [x] Expand player interaction container to send played tiles interaction messages to back-end

- [x] Receive played tile interactions from back-end to update state correctly

- [x] Implement PLAY_TILE_INTERACTION listener for error handling

- [x] Update dead pile when interaction is successful

- [x] Update player/opponent hand appropriately when interaction is successful (For next PR when meld validation is done)

- [x] Add testing

- [x] Render opponent hand that they've drawn a tile

### CODE REVIEW

File: src/pages/Game.tsx [hunk 1]
```diff
@@ -168,11 +182,96 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
+
+      const userIndex = mjGameState.getCurrentTurn();
+      // If it's user's turn, player can draw tile
+      if (mjGameState.getUsers()[userIndex].getConnectionId() === mjGameState.getMjPlayer().getConnectionId()) {
+        ws?.sendMessage(OutgoingAction.DRAW_TILE, { gameId: game.gameId });
+      } else {
+        // Change state that opponent will draw a tile
+        const opponent = mjGameState.getUsers()[userIndex] as MahjongOpponent;
+        opponent.drawTile();
+      }
```
Comment (jzc1828) [inline src/pages/Game.tsx:247]: can put `mjGameState.getUsers()[userIndex]` and `mjGameState.getMjPlayer()` into variables
```ts
      const currentMjPlayer = mjGameState.getUsers()[userIndex];
      const mjPlayer = mjGameState.getMjPlayer();
      // If it's user's turn, player can draw tile
      if (currentMjPlayer.getConnectionId() === mjPlayer.getConnectionId()) {
        ws?.sendMessage(OutgoingAction.DRAW_TILE, { gameId: game.gameId });
      } else {
        // Change state that opponent will draw a tile
        const opponent = currentMjPlayer as MahjongOpponent;
        opponent.drawTile();
      }
```

File: src/types/responses.ts [hunk 2]
```diff
@@ -135,3 +135,23 @@ export interface PlayTileJSON {
...
+
+/**
+ * Interface for PLAYED_TILE_INTERACTION payload received
+ */
+export interface PlayedTileInteractionJSON {
+  playedTiles: string[];
+  meldType: string;
+  skipInteraction: boolean;
+  success: boolean;
+}
```
Comment (jzc1828) [inline src/types/responses.ts:157]: there would be an error attribute added to the payload if `success = false`
```ts
export interface PlayedTileInteractionJSON {
  playedTiles: string[];
  meldType: string;
  skipInteraction: boolean;
  success: boolean;
  error: string;
}
```

File: src/types/responses.ts [hunk 3]
```diff
@@ -135,3 +135,23 @@ export interface PlayTileJSON {
...
+
+/**
+ * Interface for INTERACTION_SUCCESS payload received
+ */
+export interface InteractionSuccessJSON {
+  connectionId?: string;
+  playedTiles?: string[];
+  meldType?: string;
+  skipInteraction: boolean;
+}
```
Comment (jzc1828) [inline src/types/responses.ts:147]: there is also a success attribute in this payload if you ever need it:
```ts
export interface InteractionSuccessJSON {
  connectionId?: string;
  playedTiles?: string[];
  meldType?: string;
  skipInteraction: boolean;
  success: true;
}
```

File: src/pages/Game.tsx [hunk 4]
```diff
@@ -94,7 +100,7 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
       allUserEntities[currentIndex] = opponent;
       currentIndex += 1;
     }
-    player = new MahjongPlayer(current$User.username, users[indexOfCurrentUser].connectionId);
+    const player = new MahjongPlayer(current$User.username, users[indexOfCurrentUser].connectionId);
```
[the remark this answers is not in the document]
Comment (yummyblabla) [reply src/pages/Game.tsx:103]: There was a naming conflict with one of the props. Fixed to use props directly.

File: src/pages/Game.tsx (hunk 1 above)
Comment (yummyblabla) [reply src/pages/Game.tsx:247]: Okies

File: src/types/responses.ts (hunk 3 above)
Comment (yummyblabla) [reply src/types/responses.ts:147]: Is there error as well?

File: src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts [hunk 5]
```diff
@@ -129,20 +177,67 @@ class MahjongPlayer extends UserEntity {
...
-      console.log(event.target);
-      const tile = this.hand.throw();
-      if (tile !== null) {
-        callbacks[OutgoingAction.PLAY_TILE](tile.toString());
-      }
-    });
+
+    // Render play button if player has drawn.
+    if (this.hand.canPlayTile()) {
+      const playText = new PIXI.Text('PLAY TILE', PIXI_TEXT_STYLE);
```
[the remark this answers is not in the document]
Comment (yummyblabla) [reply src/modules/mahjong/MahjongPlayer/MahjongPlayer.ts:186]: Added constants in same file but not in Enums folder. Hoping to have assets to replace PIXI.Text in future

File: src/types/responses.ts (hunk 3 above)
Comment (jzc1828) [reply src/types/responses.ts:148]: nope, it would be in `PlayedTileInteractionJSON` is the lambda function failed

File: src/types/responses.ts [hunk 6]
```diff
@@ -135,3 +135,25 @@ export interface PlayTileJSON {
...
+/**
+ * Interface for PLAYED_TILE_INTERACTION payload received
+ */
+export interface PlayedTileInteractionJSON {
+  playedTiles: string[];
+  meldType: string;
+  skipInteraction: boolean;
+  success: boolean;
+  error: string;
+}
```
Comment (jzc1828) [inline src/types/responses.ts:159]: need `connectionId` here as discussed before

**Pull request merged**